lib/bitvec.cpp

```diff
@@ -36,6 +36,7 @@ std::ostream &operator<<(std::ostream &os, const bitvec &bv) {
 }
 
 bitvec &bitvec::operator>>=(size_t count) {
+    assert(static_cast<int>(count) >= 0);
```

Comment:
what is this pull request trying to solve exactly? I'm not very familiar with the `bitvec` class but this check looks a little weird and arbitrary to me. If the limit for these parameters is the maximum positive integer that fits inside an integer, maybe the parameter type should be `int` and the assert should be `assert(count > 0)`. But that seems like an arbitrary limit to me, unless I'm missing something. If the bitvec can be larger than that (unlikely to be useful in the context of this compiler), then this check makes no sense to me.